### PR TITLE
Change the structure of API views implementation

### PR DESCRIPTION
**Pull request by ahsan-ul-haq**
ECOM-2338

@zubair-arbi @aamir-khan @awais786 @tasawernawaz @clintonb please review

### CODE REVIEW

File: ecommerce/extensions/api/v2/urls.py [hunk 1]
```diff
@@ -58,8 +62,9 @@
 ]
 
 router = ExtendedSimpleRouter()
-router.register(r'courses', views.CourseViewSet) \
-    .register(r'products', views.ProductViewSet, base_name='course-product', parents_query_lookups=['course_id'])
-router.register(r'products', views.ProductViewSet)
+router.register(r'courses', course_views.CourseViewSet) \
```
Comment (zubair-arbi) [inline ecommerce/extensions/api/v2/urls.py:65]:
``` python
router.register(r'courses', course_views.CourseViewSet).register(
        r'products', product_views.ProductViewSet, base_name='course-product', parents_query_lookups=['course_id']
    )
```

File: ecommerce/extensions/api/v2/views/orders.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+            "total_excl_tax": 0.0
+        }'
+    """
+    permission_classes = (IsAuthenticated,)
+    serializer_class = serializers.OrderSerializer
+    lookup_field = AC.KEYS.ORDER_NUMBER
+
+    def get_queryset(self):
+        """
+        Returns a queryset consisting of only the authenticated user's orders.
```
Comment (zubair-arbi) [inline ecommerce/extensions/api/v2/views/orders.py]: nit: for consistency start docstring at first line:

``` python
def get_queryset(self):
    """Returns a queryset consisting of only the authenticated user's orders.
```

File: ecommerce/extensions/api/v2/views/refunds.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+from ecommerce.extensions.refund.api import find_orders_associated_with_course, create_refunds
+
+
+Refund = get_model('refund', 'Refund')
+User = get_user_model()
+
+
+class RefundCreateView(generics.CreateAPIView):
+    """
+    Creates refunds.
```
Comment (zubair-arbi) [inline ecommerce/extensions/api/v2/views/refunds.py]: nit: for consistency start docstring at first line:

File: ecommerce/extensions/api/v2/views/refunds.py [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+            refund_ids = [refund.id for refund in refunds]
+            return Response(refund_ids, status=status.HTTP_201_CREATED)
+
+        # Return HTTP 200 if we did NOT create refunds.
+        return Response([], status=status.HTTP_200_OK)
+
+
+class RefundProcessView(generics.UpdateAPIView):
+    """
+    Process--approve or deny--refunds.
```
Comment (zubair-arbi) [inline ecommerce/extensions/api/v2/views/refunds.py]: nit: for consistency start docstring at first line:

**Comment by zubair-arbi** [comment]
Just small nits otherwise code changes looks good now :+1:

**Comment by ahsan-ul-haq** [comment]
@clintonb please review.

**Comment by ahsan-ul-haq** [comment]
@rlucioni Please review.

File: ecommerce/extensions/api/v2/views/courses.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+from rest_framework.response import Response
+import waffle
+
+from ecommerce.core.constants import COURSE_ID_REGEX
+from ecommerce.courses.models import Course
+from ecommerce.extensions.api.v2.views import NonDestroyableModelViewSet
+from ecommerce.extensions.api import serializers
+
+
+Order = get_model('order', 'Order')
```
Comment (clintonb) [inline ecommerce/extensions/api/v2/views/courses.py]: None of theses are used.

**Comment by clintonb** [comment]
:+1: (once comments addressed)

File: ecommerce/extensions/api/v2/urls.py [hunk 6]
```diff
@@ -3,48 +3,52 @@
 from rest_framework_extensions.routers import ExtendedSimpleRouter
 
 from ecommerce.core.constants import COURSE_ID_PATTERN
-from ecommerce.extensions.api.v2 import views
+from ecommerce.extensions.api.v2.views import (baskets as basket_views, payments as payment_views,
+                                               orders as order_views, refunds as refund_views,
+                                               products as product_views, courses as course_views,
+                                               publications as publication_views)
```
Comment (rlucioni) [inline ecommerce/extensions/api/v2/urls.py]: Nitpick: Could you please change the name of the `publications` module to `publication`? Making it plural is strange.

File: ecommerce/extensions/api/v2/views/courses.py [hunk 7]
```diff
@@ -0,0 +1,50 @@
+"""HTTP endpoints for interacting with course courses."""
```
Comment (rlucioni) [inline ecommerce/extensions/api/v2/views/courses.py]: `"""HTTP endpoints for interacting with courses."""`

File: ecommerce/extensions/api/v2/views/publications.py [hunk 8]
```diff
@@ -0,0 +1,44 @@
+"""HTTP endpoints for interacting with course publication."""
```
Comment (rlucioni) [inline ecommerce/extensions/api/v2/views/publications.py]: `"""HTTP endpoints for course publication."""`

**Comment by clintonb** [comment]
@ahsan-ul-haq remember to delete your branches once they are merged.

**Pull request merged**